lib/mix/tasks/hexweb/check_names.ex

```diff
@@ -19,31 +16,19 @@ defmodule Mix.Tasks.Hexweb.CheckNames do
     :ok
   end
 
-  def new_packages do
-    from(p in Package, select: p.name, where: fragment("inserted_at >= CURRENT_DATE"))
-    |> HexWeb.Repo.all
-  end
-
-
-  def current_packages do
-    from(p in Package, select: p.name, where: fragment("inserted_at < CURRENT_DATE"))
-    |> HexWeb.Repo.all
-  end
-
-
-  def calculate_distances do
-    for new <- new_packages(), curr <- current_packages() do
-      [new, curr, String.jaro_distance(new, curr) |> Float.round(2)]
-    end
-  end
-
   def find_candidates(threshold) do
-    calculate_distances
-    |> Enum.filter(fn([_, _, d]) -> d > threshold end)
-    |> Enum.sort(fn([_, _, d1], [_, _, d2]) -> d1 > d2 end)
+    querystr = "SELECT pnew.name new_name, pall.name curr_name, levenshtein(pall.name, pnew.name) as dist
+                FROM packages as pall
+                CROSS JOIN packages as pnew
+                WHERE pall.name <> pnew.name
+                  AND pnew.inserted_at >= CURRENT_DATE
+                  AND levenshtein(pall.name, pnew.name) <= $1
+                ORDER BY pall.name, dist;"
```

Comment:
Since we're using a multiline queries anyway, probably a heredoc `"""` would look better :)